exercises/queen-attack/queen_attack_tests.plt

```diff
@@ -0,0 +1,62 @@
+ % Please visit http://exercism.io/languages/prolog/installing
+% for instructions on setting up prolog.
+% Visit http://exercism.io/languages/prolog/tests
+% for help running the tests for prolog exercises.
+
+% The goal below allows tests to be skipped
+% unless the "--all" flag is passed at
+% the command line.
+
+pending :-
+    current_prolog_flag(argv, ['--all'|_]).
+pending :-
+    write('\nA TEST IS PENDING!\n'),
+    fail.
+
+:- begin_tests(create_tests).
+
+    test(create_in_center, condition(true)) :-
+      create((3,3)).
+
+    test(valid_position, condition(pending)) :-
+      create((2,2)).
+
+    test(must_have_positive_row, [fail, condition(pending)]) :-
+      create((-2,2)).
+
+    test(row_smaller_than_board_size, [fail, condition(pending)]) :-
+      create((8,4)).
+
+    test(must_have_positive_column, [fail, condition(pending)]) :-
+      create((2,-2)).
+
+    test(column_smaller_than_board_size, [fail, condition(pending)]) :-
+      create((4,8)).
+
+:- end_tests(create_tests).
+
+
+:- begin_tests(attack_tests).
+
+    test(cant_attack, [fail, condition(pending)]) :-
+      attack((2,4), (6,6)).
+
+    test(attack_on_same_row, condition(pending)) :-
+      attack((2,4), (2,6)).
+
+    test(attack_same_column, condition(pending)) :-
+      attack((4,5), (2,5)).
+
+    test(attack_first_diagonal, condition(pending)) :-
+      attack((2,2), (0,4)).
+
+    test(attack_second_diagonal, condition(pending)) :-
+      attack((2,2), (3,1)).
+
+    test(attack_third_diagonal, condition(pending)) :-
+      attack((2,2), (1,1)).
+
+    test(attack_fourth_diagonal, condition(pending)) :-
+      attack((2,2), (5,5)).
+
```

Comment:
Should we add tests for queens on the same tile or queens off the board? It seems like a fail might be appropriate? I defer to those with more experience with the idioms of prolog code.

Reply:
The test cases are from [here](https://github.com/exercism/problem-specifications/blob/master/exercises/queen-attack/canonical-data.json), in order to add a case here, first it should be there.

Reply:
From the comments field of the spec: 
>"Some languages implement tests beyond this set, such as checking for two pieces being placed on the same position, representing the board graphically, or using standard chess notation. Those tests can be offered as extra credit."

Reply:
So, what should return ```attack((x,y),(x,y))``` in your opinion?

Reply:
This is where I'd want someone with actual prolog experience to comment, because it shouldn't return `true`, but in my opinion false doesn't give you enough information.

Throwing exceptions seems like a "good" idea, but I can't find a good resource on how to test for that.

Possibly it would look like this:
```prolog
attack(A, A) :- !,
  throw(domain_error("Two queens cannot occupy the same space.", A)).
```

Reply:
maybe, if two queens are in the same spot, stands for one being eaten

Reply:
What do you think @parkerl ?